bindings/spt/platform.c

```diff
@@ -60,6 +60,15 @@ int platform_set_tls_base(uint64_t base)
 #elif defined(__aarch64__)
     cpu_set_tls_base(base);
     return 0;
+#elif defined(__powerpc64__)
+    /* first variable in TLS will be accessed using -0x7000(r13) */
+    base += 0x7000;
```

Comment:
Why is this magic constant here? Where did you get it? Why should this behave any differently from the other architecture's code paths, which do not use any magic constants?

Reply:
Again, the offset of the TP as in that Figure 3.1. I can introduce a #define if you wish.

Reply:
Well, I think that `solo5_set_tls_base()` should do what it says on the tin and nothing more. In other words, this shouldn't be here, and it should be up to the caller to add any magic offsets.

Reply:
I modified it now to this:

```
+#elif defined(__powerpc64__)
+    /* first variable in TLS will be accessed using -0x7000(r13) */
+    cpu_set_tls_base(base + 0x7000);
+    return 0;
```

Reply:
I think we're misunderstanding each other. The caller in the context of this change is `test_tls`, that's where the magic values should go. The bindings should just take whatever value is passed to `solo5_set_tls_base()` and pass it all the way through.

Reply:
I wasn't sure about this. It didn't feel right to adjust the address on the level of the test case. I though on that level it should be transparent and the CPU specific artifact with the offset would be adjusted further down ...